fleetspeak/src/e2etesting/lib/setup_components.go

```diff
@@ -0,0 +1,296 @@
+
+	// Client services configuration
+	client_service_conf := spb.ClientServiceConfig{Name: "FRR_client", Factory: "Daemon"}
+	var payload daemonservice_pb.Config
+	payload.Argv = append(payload.Argv, "python", "../../../../frr_python/frr_client.py")
+	client_service_conf.Config, _ = ptypes.MarshalAny(&payload)
+	os.Mkdir(path.Join(temp_path, "textservices"), 0777)
+	os.Mkdir(path.Join(temp_path, "services"), 0777)
+	os.Create(path.Join(temp_path, "communicator.txt"))
+	ioutil.WriteFile(path.Join(temp_path, "textservices", "frr.textproto"), []byte(proto.MarshalTextString(&client_service_conf)), 0644)
+
+	// Server services configuration
+	server_service_conf := services_pb.ServiceConfig{Name: "FRR_server", Factory: "GRPC"}
+	grpc_config := grpcservice_pb.Config{Target: fmt.Sprintf("localhost:%v", fs_frontend_port), Insecure: true}
+	server_service_conf.Config, _ = ptypes.MarshalAny(&grpc_config)
+	server_conf := services_pb.ServerConfig{Services: []*services_pb.ServiceConfig{&server_service_conf}, BroadcastPollTime: new(duration.Duration)}
+	server_conf.BroadcastPollTime.Seconds = 1
+	built_server_services_config_path := path.Join(temp_path, "server.services.config")
+	ioutil.WriteFile(built_server_services_config_path, []byte(proto.MarshalTextString(&server_conf)), 0644)
+}
+
+func startProcesses(started_processes *startedProcessesPids, temp_path string, fs_frontend_port, fs_admin_port int) {
+	// Start server
+	server_run_cmd := exec.Command("server", "-logtostderr", "-components_config", path.Join(temp_path, "server.config"), "-services_config", path.Join(temp_path, "server.services.config"))
+	go startProcess(server_run_cmd, started_processes.server_pid)
+
+	server_start_time := time.Now()
+
+	// Sleep and start client
+	time.Sleep(time.Second * 1)
+	client_run_cmd := exec.Command("client", "-logtostderr", "-config", path.Join(temp_path, "linux_client.config"))
+	go startProcess(client_run_cmd, started_processes.client_pid)
+
+	// Get new client's id and start service in current process
+	client_id := getNewClientId(fs_admin_port, server_start_time)
+	fmt.Println("Enrolled client_id: ", client_id)
+	service_run_cmd := exec.Command(
+		"python",
+		"../../../../frr_python/frr_server.py",
+		fmt.Sprintf("--client_id=%v", client_id),
+		fmt.Sprintf("--fleetspeak_message_listen_address=localhost:%v", fs_frontend_port),
+		fmt.Sprintf("--fleetspeak_server=localhost:%v", fs_admin_port))
+	startProcess(service_run_cmd, started_processes.service_pid)
+}
+
+func main() {
+	flag.Parse()
+
+	temp_path, _ := ioutil.TempDir(os.TempDir(), "*_fleetspeak")
+	fs_frontend_port := 6062
+	fs_admin_port := 6061
+
+	configureFleetspeak(temp_path, fs_frontend_port, fs_admin_port)
+
+	var started_processes startedProcessesPids
+	started_processes.init()
+	defer started_processes.killAll()
+
+	startProcesses(&started_processes, temp_path, fs_frontend_port, fs_admin_port)
```

Comment:
Should this be started_process.start()? (I.e. should this be a "member function" of startedProcessPids?